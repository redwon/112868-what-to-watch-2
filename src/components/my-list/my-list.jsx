import React from 'react';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import {MovieType} from '../../types';

import Header from '../header/header';
import Footer from '../footer/footer';
import MoviesList from '../movies-list/movies-list';

const MyList = ({movies}) => {
  if (!movies) {
    return null;
  }

  return (
    <div className="user-page">
      <Header className="user-page__head">
        <h1 className="page-title user-page__title">My list</h1>
      </Header>
      <section className="catalog">
        <h2 className="catalog__title visually-hidden">Catalog</h2>
        <MoviesList movies={movies} />
      </section>
      <Footer />
    </div>
  );
};

MyList.propTypes = {
  movies: PropTypes.arrayOf(MovieType),
};

const mapStateToProps = (state, props) => Object.assign({}, props, {
  movies: state.favoriteMovies,
});

export {MyList};

export default connect(mapStateToProps)(MyList);
