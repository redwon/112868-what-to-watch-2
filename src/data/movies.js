const movies = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`
  },
  {
    name: `Bohemian Rhapsody`,
    img: `img/bohemian-rhapsody.jpg`
  },
  {
    name: `Macbeth`,
    img: `img/macbeth.jpg`
  },
  {
    name: `Mindhunter`,
    img: `img/mindhunter.jpg`
  }
];

export default movies;
