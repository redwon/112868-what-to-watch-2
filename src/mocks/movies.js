const movies = [
  {
    id: 1,
    name: `War of the Worlds`,
    posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg`,
    previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg`,
    backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg`,
    backgroundColor: `#9B7E61`,
    description: `Lorem ipsum`,
    rating: 9.3,
    scoresCount: 386834,
    director: `Steven Spielberg`,
    starring: [`Tom Cruise`, `Dakota Fanning`, `Tim Robbins`],
    runTime: 116,
    genre: `Adventure`,
    released: 2005,
    isFavorite: false,
    videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: 2,
    name: `Bronson`,
    posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/bronson.jpg`,
    previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/bronson.jpg`,
    backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/bronson.jpg`,
    backgroundColor: `#73B39A`,
    description: `Lorem ipsum`,
    rating: 3.6,
    scoresCount: 109661,
    director: `Nicolas Winding Refn`,
    starring: [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    runTime: 92,
    genre: `Action`,
    released: 2008,
    isFavorite: false,
    videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

export default movies;
