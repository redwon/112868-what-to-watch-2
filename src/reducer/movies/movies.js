export const initialState = [];

export const ActionType = {
  LOAD_MOVIES: `LOAD_MOVIES`,
};

export const ActionCreator = {
  loadMovies: (movies) => ({
    type: ActionType.LOAD_MOVIES,
    payload: movies
  }),
};

export const Operations = {
  loadMovies: () => (dispatch, _, api) => {
    return api.get(`/films`)
      .then((response) => {
        dispatch(ActionCreator.loadMovies(response.data));
      });
  }
};

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.LOAD_MOVIES:
      return action.payload;
  }

  return state;
};
