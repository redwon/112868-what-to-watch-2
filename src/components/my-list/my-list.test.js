import React from 'react';
import renderer from 'react-test-renderer';

import movies from '../../mocks/movies';

import {MyList} from './my-list';

jest.mock(`../header/header`, () => () => `Header`);

it(`renders correctly`, () => {
  const tree = renderer
    .create(<MyList
      movies={movies}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
