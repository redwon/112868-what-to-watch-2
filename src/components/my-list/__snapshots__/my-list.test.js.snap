// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="user-page"
>
  Header
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            className="player__video"
            muted={true}
            poster="https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="#"
            onClick={[Function]}
          >
            War of the Worlds
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            className="player__video"
            muted={true}
            poster="https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/bronson.jpg"
            src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
          />
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="#"
            onClick={[Function]}
          >
            Bronson
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 
        2019
         What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
